File: server/Models/Domain/ProjectMember.cs
using server.Models.Domain.Enums;

namespace server.Models.Domain;

public record class ProjectMember
{
    public int ProjectMemberId { get; init; }
    public ProjectRole Role { get; init; }
    public DateTime InvitedAt { get; init; }
    public DateTime? JoinedAt { get; init; }

    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }

    // Foreign Keys
    public int ProjectId { get; init; }
    public string UserId { get; init; } = string.Empty;

    // Navigation Properties
    public virtual Project Project { get; init; } = null!;
    public virtual ApplicationUser User { get; init; } = null!;
}

File: frontend/src/types/projectMember/projectMember.ts
import type { ProjectRole } from "../project"

export interface ProjectMember {
    id: number;
    role: ProjectRole;
    invitedAt: string;
    joinedAt?: string;
    createdAt: string;
    updatedAt: string;
    projectId: number;
    email: string;
    userName?: string;
}

export interface InviteMemberRequest {
    email: string;
    role: ProjectRole;
}